### DISCUSSION TITLE
MFA google authenticator

### DISCUSSION THREAD

**Discussion by PaS3** [keepassxreboot/keepassxc#6679] [Ideas]
Can an "authenticator" be integrated in the next build, specifically unlocking a database with MFA?

I use this MFA combo with SSH on Linux Debian 10
https://duo.com/product/multi-factor-authentication-mfa/duo-mobile-app
https://github.com/google/google-authenticator-libpam

C# version of KeePass runs something on network stack, kinda creepy...
$      mono /opt/KeePass-2.48.1/KeePass.exe
Local Address           Foreign Address         State       PID/Program name     Timer
tcp        0      0 127.0.0.1:56031         0.0.0.0:*               LISTEN      24711/mono

KeePassXC does not do much on the network.

**Comment by droidmonkey** [comment]
No this is not possible with an offline database. This is discussed elsewhere on the issues board.

**Reply by PaS3** [reply]
I capture a QR code and it generates unique 6 numbers without going anywhere on the internet, the other end of verification can be stored as private keys with the database that is then encrypted.  No part of this second factor authentication is actually on the internet.

**Reply by droidmonkey** [reply]
That is not how it works, please trust us we know this inside and out.

**Reply by PaS3** [reply]
I need to review my SSH / PAM-auth communications again. I was fairly certain I did not see any connections on the back end.

**Reply by droidmonkey** [reply]
Wasn't talking about the internet aspect, just how totp actually works and what information is required to be stored and where/how to make it work.
